### PR TITLE
[NCL-4652]: Update orchestrator to add extra field for scmTag

### PR DESCRIPTION
**Pull request by jbartece**
### Checklist:

* [ ] Have you added a note in the CHANGELOG.md for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

**Comment by thescouser89** [comment]
Are we going to migration the existing scmRevision information (containing the tags) to scmTag, and leave scmRevision empty / repopulated with commit id?

File: integration-test/src/test/java/org/jboss/pnc/integration/BuildTasksRestTest.java [hunk 1]
```diff
@@ -92,7 +92,7 @@ public void shouldTriggerBuildExecution() {
 
         BuildExecutionConfiguration buildExecutionConfig = BuildExecutionConfiguration.build(
                 1, "test-content-id", 1, "mvn clean install", "jboss-modules", "scm-url", "master",
-                "origin-scm-url", false, "dummy-docker-image-id", "dummy.repo.url/repo", SystemImageType.DOCKER_IMAGE, false, null, new HashMap<>(),
+                "1.0.0.redhat-1", "origin-scm-url", false, "dummy-docker-image-id", "dummy.repo.url/repo", SystemImageType.DOCKER_IMAGE, false, null, new HashMap<>(),
```
Comment (janinko) [inline integration-test/src/test/java/org/jboss/pnc/integration/BuildTasksRestTest.java]: Shouldn't the previous parameter scmRevision "master" be now some sha?

File: pnc-mock/src/main/java/org/jboss/pnc/mock/executor/BuildExecutionConfigurationMock.java [hunk 2]
```diff
@@ -55,6 +56,7 @@ public static BuildExecutionConfiguration mockConfig() {
         mock.setBuildScript("mvn install");
         mock.setScmRepoURL("http://www.github.com");
         mock.setScmRevision("1234567890");
```
Comment (janinko) [inline pnc-mock/src/main/java/org/jboss/pnc/mock/executor/BuildExecutionConfigurationMock.java]: Shouldn't the SCM revision have sha format?

File: pnc-mock/src/main/java/org/jboss/pnc/mock/spi/BuildExecutionConfigurationMock.java [hunk 3]
```diff
@@ -38,6 +38,7 @@ public static BuildExecutionConfiguration mock() {
                 "configuration name",
                 "https://pathToRepo.git",
                 "1111111",
```
Comment (janinko) [inline pnc-mock/src/main/java/org/jboss/pnc/mock/spi/BuildExecutionConfigurationMock.java]: Shouldn't the SCM revision have sha format?

File: rest-model/src/test/java/org/jboss/pnc/restmodel/serialization/BuildExecutionConfigurationTest.java [hunk 4]
```diff
@@ -47,6 +47,7 @@ public void serializeAndDeserializeBuildResult() throws IOException, BuildDriver
                 "configuration name",
                 "https://pathToRepo.git",
                 "1111111",
```
Comment (janinko) [inline rest-model/src/test/java/org/jboss/pnc/restmodel/serialization/BuildExecutionConfigurationTest.java]: Shouldn't the SCM revision have sha format?

Comment (jbartece) [reply rest-model/src/test/java/org/jboss/pnc/restmodel/serialization/BuildExecutionConfigurationTest.java]: Yes, it should. I'll fix it.

File: model/src/main/resources/db-update/00006-from-1.5.1-to-1.6.0.sql [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+-- limitations under the License.
+--
+
+
+-- [NCL-4652]
+-- Copy scmRevision to scmTag for old builds
+alter table buildrecord add column scmTag varchar(255);
+
+UPDATE buildrecord
+SET scmTag = subquery.scmRevision
```
Comment (pkocandr) [inline model/src/main/resources/db-update/00006-from-1.5.1-to-1.6.0.sql]: This seems very strange to me. Couldn't this be just
```UPDATE buildrecord SET scmTag = scmRevision;```
It assigns the scmRevision value to scmTag from the same row. Or wouldn't that worked for some reason?

Comment (jbartece) [reply model/src/main/resources/db-update/00006-from-1.5.1-to-1.6.0.sql]: Honestly, I have never tried such update. I can test it out.

Comment (jbartece) [reply model/src/main/resources/db-update/00006-from-1.5.1-to-1.6.0.sql]: You were right. It was enough. Simplified. Merging

**Pull request merged**